scheduler/tasks/exporter-task/src/main/kotlin/uk/ac/ebi/scheduler/pmc/exporter/service/PmcExporterService.kt

```diff
@@ -27,40 +24,55 @@ class PmcExporterService(
 ) {
     private val pmcDataIterator = pageIterator(pmcRepository)
 
-    suspend fun exportPmcLinks() = withContext(Dispatchers.IO) {
+    fun exportPmcLinks() {
         logger.info { "Started exporting PMC links" }
 
         ftpClient.login()
 
         pmcDataIterator
             .asSequence()
-            .mapIndexed { part, page -> launch { writeLinks(part + 1, page.content) } }
+            .mapIndexed { part, page -> writeLinks(part + 1, page.content) }
             .toList()
-            .joinAll()
 
         ftpClient.logout()
         ftpClient.disconnect()
 
         logger.info { "Finished exporting PMC links" }
     }
 
-    private suspend fun writeLinks(part: Int, submissions: List<PmcData>) {
+    private fun writeLinks(part: Int, submissions: List<PmcData>) {
         logger.info { "Exporting PMC links part $part" }
         val links = submissions.map { it.toLink() }
-        withContext(Dispatchers.IO) { write(part, links) }
+
+        write(part, links)
     }
 
     private fun write(part: Int, links: List<Link>) {
-        logger.info { "writing file part $part" }
-        val xml = xmlWriter.writeValueAsString(Links(links))
+        logger.info { "Writing file part $part" }
+
+        val xml = xmlWriter.writeValueAsString(Links(links)).byteInputStream()
         val path = "${appProperties.outputPath}/${String.format(appProperties.fileName, part)}"
-        ftpClient.storeFile(path, xml.byteInputStream())
+
+        try {
+            ftpClient.storeFile(path, xml)
```

Comment:
I would suggest writing a wrapper class over apache FTPClient, like our own class that handles retry so we can keep the logic here clean here.  (hight specialization)